client/consensus/beefy/src/lib.rs

```diff
@@ -512,19 +504,34 @@ where
 	Err(ClientError::Backend(err_msg))
 }
 
-fn expect_validator_set<B, R>(
+fn expect_validator_set<B, BE, R>(
 	runtime: &R,
-	at_hash: B::Hash,
+	backend: &BE,
+	at_header: &B::Header,
 ) -> ClientResult<ValidatorSet<AuthorityId>>
 where
 	B: Block,
+	BE: Backend<B>,
 	R: ProvideRuntimeApi<B>,
 	R::Api: BeefyApi<B, AuthorityId>,
 {
 	runtime
 		.runtime_api()
-		.validator_set(at_hash)
+		.validator_set(at_header.hash())
 		.ok()
 		.flatten()
-		.ok_or_else(|| ClientError::Backend("BEEFY pallet expected to be active.".into()))
+		.or_else(|| {
+			// if state unavailable, fallback to walking up the chain looking for the header
+			// Digest emitted when validator set active 'at_header' was enacted.
+			let blockchain = backend.blockchain();
+			let mut header = at_header.clone();
```

Comment:
Nit: I think we don't need to clone here.

Reply:
we do because `at_header` is reference, whereas `header` needs to be owned because `expect_header()` below returns owned object

Reply:
You're right. Sorry.